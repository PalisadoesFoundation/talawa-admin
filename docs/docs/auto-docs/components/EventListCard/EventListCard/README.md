[**talawa-admin**](../../../README.md)

***

[talawa-admin](../../../README.md) / components/EventListCard/EventListCard

# components/EventListCard/EventListCard

## Interfaces

- [InterfaceEventListCardProps](interfaces/InterfaceEventListCardProps.md)

## Functions

- [default](functions/default.md)
