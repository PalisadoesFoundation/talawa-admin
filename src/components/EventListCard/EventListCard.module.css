.cards h2 {
  font-size: 15px;
  color: #fff;
}
.cards > h3 {
  font-size: 17px;
}
.cards > p {
  font-size: 14px;
  margin-top: 0px;
  margin-bottom: 7px;
}
.cards a {
  color: #fff;
  font-weight: 600;
}
.cards a:hover {
  color: black;
}
.cards {
  position: relative;
  overflow: hidden;
  transition: all 0.3s;
}
.dispflex {
  display: flex;
  height: 50px;
  transition: transform 0.3s ease;
  cursor: pointer;
}
.cards:hover {
  transform: scale(2.5);
  z-index: 5;
}
.cards:hover h2 {
  font-size: 0.4vmax;
  margin-bottom: 0;
}
.iconContainer {
  display: flex;
  justify-content: flex-end;
}
.icon {
  margin: 1px;
}

.cards {
  width: 100%;
  background: rgba(40, 167, 69, 1);
  padding: 2px 3px;
  border-radius: 5px;
  border: 1px solid #e8e8e8;
  box-shadow: 0 3px 5px #c9c9c9;
  color: #737373;
  box-sizing: border-box;
}
.cards:last-child:nth-last-child(odd) {
  grid-column: auto / span 2;
}

.cards:first-child:nth-last-child(even),
.cards:first-child:nth-last-child(even) ~ .box {
  grid-column: auto / span 1;
}

.sidebarsticky > input {
  text-decoration: none;
  margin-bottom: 50px;
  border-color: #e8e5e5;
  width: 80%;
  border-radius: 7px;
  padding-top: 5px;
  padding-bottom: 5px;
  padding-right: 10px;
  padding-left: 10px;
  box-shadow: none;
}
.datediv {
  display: flex;
  flex-direction: row;
  margin-bottom: 15px;
}
.datebox {
  width: 90%;
  border-radius: 7px;
  border-color: #e8e5e5;
  outline: none;
  box-shadow: none;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-right: 5px;
  padding-left: 5px;
}
.checkboxdiv > label {
  margin-right: 50px;
}
.checkboxdiv > label > input {
  margin-left: 10px;
}
.dispflex {
  display: flex;
  margin-bottom: 5px;
  margin-right: 5px;
}
.dispflex > input {
  width: 20%;
  border: none;
  box-shadow: none;
  margin-top: 5px;
}
.checkboxdiv {
  display: flex;
}
.checkboxdiv > div {
  width: 50%;
}

.preview {
  display: flex;
  flex-direction: row;
  font-weight: 900;
  font-size: 16px;
  color: rgb(80, 80, 80);
}
.view {
  margin-left: 2%;
  font-weight: 600;
  font-size: 16px;
  color: #707070;
}
.flexdir {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border: none;
}

.form_wrapper {
  margin-top: 27px;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  position: absolute;
  display: flex;
  flex-direction: column;
  padding: 20px 30px;
  background: #ffffff;
  border-color: #e8e5e5;
  border-width: 5px;
  border-radius: 10px;
  width: 40%;
}

.form_wrapper form {
  display: flex;
  align-items: left;
  justify-content: left;
  flex-direction: column;
}
.titlemodal {
  color: #707070;
  font-weight: 600;
  font-size: 20px;
  margin-bottom: 20px;
  padding-bottom: 5px;
  border-bottom: 3px solid #31bb6b;
  width: 65%;
}
.cancel > i {
  margin-top: 5px;
  transform: scale(1.2);
  cursor: pointer;
  color: #707070;
}
.modalbody {
  width: 50px;
}
.list_box {
  height: 70vh;
  overflow-y: auto;
  width: auto;
}
@media only screen and (max-width: 600px) {
  .form_wrapper {
    width: 90%;
    top: 45%;
  }
}

.customButton {
  background-color: #31bb6b;
  border: #31bb6b;
}
