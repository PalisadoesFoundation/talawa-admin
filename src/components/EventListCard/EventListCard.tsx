import type { ChangeEvent } from 'react';
import React, { useEffect, useState } from 'react';
import { useMutation } from '@apollo/client';
import { toast } from 'react-toastify';
import { useTranslation } from 'react-i18next';
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';
import styles from './EventListCard.module.css';
import {
  DELETE_EVENT_MUTATION,
  UPDATE_EVENT_MUTATION,
} from 'GraphQl/Mutations/mutations';
import { Form } from 'react-bootstrap';
import { errorHandler } from 'utils/errorHandler';
import { useHistory } from 'react-router-dom';

interface InterfaceEventListCardProps {
  key: string;
  id: string;
  eventLocation: string;
  eventName: string;
  eventDescription: string;
  regDate: string;
  regEndDate: string;
  startTime: string;
  endTime: string;
  allDay: boolean;
  recurring: boolean;
  isPublic: boolean;
  isRegisterable: boolean;
}
function eventListCard(props: InterfaceEventListCardProps): JSX.Element {
  const { t } = useTranslation('translation', {
    keyPrefix: 'eventListCard',
  });
  const [eventmodalisOpen, setEventModalIsOpen] = useState(false);
  const [alldaychecked, setAllDayChecked] = useState(true);
  const [recurringchecked, setRecurringChecked] = useState(false);
  const [publicchecked, setPublicChecked] = useState(true);
  const [registrablechecked, setRegistrableChecked] = React.useState(false);
  const [eventDeleteModalIsOpen, setEventDeleteModalIsOpen] = useState(false);
  const [eventUpdateModalIsOpen, setEventUpdateModalIsOpen] = useState(false);
  const history = useHistory();
  const [formState, setFormState] = useState({
    title: '',
    eventdescrip: '',
    location: '',
    startTime: '08:00:00',
    endTime: '18:00:00',
  });
  const showViewModal = (): void => {
    setEventModalIsOpen(true);
  };
  const hideViewModal = (): void => {
    setEventModalIsOpen(false);
  };

  const toggleDeleteModal = (): void => {
    setEventDeleteModalIsOpen(!eventDeleteModalIsOpen);
  };

  const toggleUpdateModel = (): void => {
    setEventUpdateModalIsOpen(!eventUpdateModalIsOpen);
  };

  useEffect(() => {
    setFormState({
      title: props.eventName,
      eventdescrip: props.eventDescription,
      location: props.eventLocation,
      startTime: props.startTime?.split('.')[0] || '08:00:00',
      endTime: props.endTime?.split('.')[0] || '18:00:00',
    });

    setAllDayChecked(props.allDay);
    setRecurringChecked(props.recurring);
    setPublicChecked(props.isPublic);
    setRegistrableChecked(props.isRegisterable);
  }, []);

  const [create] = useMutation(DELETE_EVENT_MUTATION);
  const [updateEvent] = useMutation(UPDATE_EVENT_MUTATION);

  const deleteEvent = async (): Promise<void> => {
    try {
      const { data } = await create({
        variables: {
          id: props.id,
        },
      });

      /* istanbul ignore next */
      if (data) {
        toast.success(t('eventDeleted'));
        setTimeout(() => {
          window.location.reload();
        }, 2000);
      }
    } catch (error: any) {
      /* istanbul ignore next */
      errorHandler(t, error);
    }
  };

  const updateEventHandler = async (
    e: ChangeEvent<HTMLFormElement>
  ): Promise<void> => {
    e.preventDefault();

    try {
      const { data } = await updateEvent({
        variables: {
          id: props.id,
          title: formState.title,
          description: formState.eventdescrip,
          isPublic: publicchecked,
          recurring: recurringchecked,
          isRegisterable: registrablechecked,
          allDay: alldaychecked,
          location: formState.location,
          startTime: !alldaychecked ? formState.startTime + 'Z' : null,
          endTime: !alldaychecked ? formState.endTime + 'Z' : null,
        },
      });

      /* istanbul ignore next */
      if (data) {
        toast.success(t('eventUpdated'));
        setTimeout(() => {
          window.location.reload();
        }, 2000);
      }
    } catch (error: any) {
      /* istanbul ignore next */
      errorHandler(t, error);
    }
  };

  const openEventDashboard = (): void => {
    history.push(`/event/${props.id}`);
  };

  return (
    <>
      <div className="">
        <div
          className={styles.cards}
          onClick={showViewModal}
          data-testid="card"
        >
          <div className={styles.dispflex}>
            <h2>
              {props.eventName ? (
                <>
                  {props.eventName.length > 150 ? (
                    <>{props.eventName.substring(0, 150)}...</>
                  ) : (
                    <>{props.eventName}</>
                  )}
                </>
              ) : (
                <>Dogs Care</>
              )}
            </h2>
          </div>
        </div>
      </div>
      {/* preview modal */}
      <Modal show={eventmodalisOpen}>
        <Modal.Header>
          <p className={styles.titlemodal}>{t('eventDetails')}</p>
          <Button
            onClick={hideViewModal}
            data-testid="createEventModalCloseBtn"
          >
            <i className="fa fa-times"></i>
          </Button>
        </Modal.Header>
        <Modal.Body>
          <Form>
            <div>
              <p className={styles.preview}>
                {t('eventTitle')}:{' '}
                <span className={styles.view}>
                  {props.eventName ? (
                    <>
                      {props.eventName.length > 100 ? (
                        <>{props.eventName.substring(0, 100)}...</>
                      ) : (
                        <>{props.eventName}</>
                      )}
                    </>
                  ) : (
                    <>Dogs Care</>
                  )}
                </span>
              </p>
              <p className={styles.preview}>
                {t('location')}:
                <span className={styles.view}>
                  {props.eventLocation ? (
                    <>{props.eventLocation}</>
                  ) : (
                    <>India</>
                  )}
                </span>
              </p>
              <p className={styles.preview}>
                {t('description')}:{' '}
                <span className={styles.view}>
                  {props.eventDescription && props.eventDescription.length > 256
                    ? props.eventDescription.substring(0, 256) + '...'
                    : props.eventDescription}
                </span>
              </p>
              <p className={styles.preview}>
                {t('on')}: <span className={styles.view}>{props.regDate}</span>
              </p>
              <p className={styles.preview}>
                {t('end')}:{' '}
                <span className={styles.view}>{props.regEndDate}</span>
              </p>
              <Button
                className={styles.customButton}
                variant="success"
                onClick={openEventDashboard}
              >
                {' '}
                Show Event Dashboard{' '}
              </Button>
            </div>
            <div className={styles.iconContainer}>
              <Button
                size="sm"
                data-testid="editEventModalBtn"
                className={styles.icon}
                onClick={toggleUpdateModel}
              >
                {' '}
                <i className="fas fa-edit"></i>
              </Button>
              <Button
                size="sm"
                data-testid="deleteEventModalBtn"
                className={styles.icon}
                onClick={toggleDeleteModal}
              >
                {' '}
                <i className="fa fa-trash"></i>
              </Button>
            </div>
          </Form>
        </Modal.Body>
      </Modal>

      {/* delete modal */}
      <Modal
        size="sm"
        id={`deleteEventModal${props.id}`}
        show={eventDeleteModalIsOpen}
        onHide={toggleDeleteModal}
        backdrop="static"
        keyboard={false}
      >
        <Modal.Header closeButton className="bg-primary">
          <Modal.Title
            className="text-white"
            id={`deleteEventModalLabel${props.id}`}
          >
            {t('deleteEvent')}
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>{t('deleteEventMsg')}</Modal.Body>
        <Modal.Footer>
          <Button
            type="button"
            className="btn btn-danger"
            data-dismiss="modal"
            onClick={toggleDeleteModal}
            data-testid="EventDeleteModalCloseBtn"
          >
            {t('no')}
          </Button>
          <Button
            type="button"
            className="btn btn-success"
            onClick={deleteEvent}
            data-testid="deleteEventBtn"
          >
            {t('yes')}
          </Button>
        </Modal.Footer>
      </Modal>

      {/* Edit Modal */}
      <Modal
        id={`editEventModal${props.id}`}
        show={eventUpdateModalIsOpen}
        onHide={toggleUpdateModel}
        backdrop="static"
        keyboard={false}
      >
        <Modal.Header closeButton className="bg-primary">
          <Modal.Title
            id={`editEventModal${props.id}Label`}
            className="text-white"
          >
            {' '}
            {t('editEvent')}
          </Modal.Title>
        </Modal.Header>
        <form onSubmit={updateEventHandler}>
          <Modal.Body>
            <label htmlFor="eventtitle">{t('eventTitle')}</label>
            <Form.Control
              type="title"
              id="eventitle"
              className="mb-3"
              autoComplete="off"
              data-testid="updateTitle"
              required
              value={formState.title}
              onChange={(e): void => {
                setFormState({
                  ...formState,
                  title: e.target.value,
                });
              }}
            />
            <label htmlFor="eventdescrip">{t('description')}</label>
            <Form.Control
              type="eventdescrip"
              id="eventdescrip"
              className="mb-3"
              autoComplete="off"
              data-testid="updateDescription"
              required
              value={formState.eventdescrip}
              onChange={(e): void => {
                setFormState({
                  ...formState,
                  eventdescrip: e.target.value,
                });
              }}
            />
            <label htmlFor="eventLocation">{t('location')}</label>
            <Form.Control
              type="text"
              id="eventLocation"
              className="mb-3"
              autoComplete="off"
              data-testid="updateLocation"
              required
              value={formState.location}
              onChange={(e): void => {
                setFormState({
                  ...formState,
                  location: e.target.value,
                });
              }}
            />
            {!alldaychecked && (
              <div className={styles.datediv}>
                <div className="me-1">
                  <label htmlFor="startTime">{t('startTime')}</label>
                  <Form.Control
                    id="startTime"
                    value={formState.startTime}
                    data-testid="updateStartTime"
                    onChange={(e): void =>
                      setFormState({
                        ...formState,
                        startTime: e.target.value,
                      })
                    }
                  />
                </div>
                <div className="ms-1">
                  <label htmlFor="endTime">{t('endTime')}</label>
                  <Form.Control
                    id="endTime"
                    value={formState.endTime}
                    data-testid="updateEndTime"
                    onChange={(e): void =>
                      setFormState({
                        ...formState,
                        endTime: e.target.value,
                      })
                    }
                  />
                </div>
              </div>
            )}
            <div className={styles.checkboxdiv}>
              <div className={styles.dispflex}>
                <label htmlFor="allday">{t('allDay')}?</label>
                <Form.Switch
                  id="allday"
                  type="checkbox"
                  data-testid="updateAllDay"
                  checked={alldaychecked}
                  onChange={(): void => setAllDayChecked(!alldaychecked)}
                />
              </div>
              <div className={styles.dispflex}>
                <label htmlFor="recurring">{t('recurringEvent')}:</label>
                <Form.Switch
                  id="recurring"
                  type="checkbox"
                  data-testid="updateRecurring"
                  checked={recurringchecked}
                  onChange={(): void => setRecurringChecked(!recurringchecked)}
                />
              </div>
            </div>
            <div className={styles.checkboxdiv}>
              <div className={styles.dispflex}>
                <label htmlFor="ispublic">{t('isPublic')}?</label>
                <Form.Switch
                  id="ispublic"
                  type="checkbox"
                  data-testid="updateIsPublic"
                  checked={publicchecked}
                  onChange={(): void => setPublicChecked(!publicchecked)}
                />
              </div>
              <div className={styles.dispflex}>
                <label htmlFor="registrable">{t('isRegistrable')}?</label>
                <Form.Switch
                  id="registrable"
                  type="checkbox"
                  data-testid="updateRegistrable"
                  checked={registrablechecked}
                  onChange={(): void =>
                    setRegistrableChecked(!registrablechecked)
                  }
                />
              </div>
            </div>
          </Modal.Body>
          <Modal.Footer>
            <Button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
              data-testid="EventUpdateModalCloseBtn"
              onClick={toggleUpdateModel}
            >
              {t('close')}
            </Button>
            <Button
              type="submit"
              className="btn btn-success"
              data-testid="updatePostBtn"
            >
              {t('updatePost')}
            </Button>
          </Modal.Footer>
        </form>
      </Modal>
    </>
  );
}
export {};
export default eventListCard;
