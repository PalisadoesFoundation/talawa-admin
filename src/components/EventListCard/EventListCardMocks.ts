import {
  DELETE_EVENT_MUTATION,
  REGISTER_EVENT,
  UPDATE_EVENT_MUTATION,
} from 'GraphQl/Mutations/mutations';

export const MOCKS = [
  {
    request: {
      query: DELETE_EVENT_MUTATION,
      variables: { id: '1' },
    },
    result: {
      data: {
        removeEvent: {
          _id: '1',
        },
      },
    },
  },
  {
    request: {
      query: DELETE_EVENT_MUTATION,
      variables: { id: '1', recurringEventDeleteType: 'thisInstance' },
    },
    result: {
      data: {
        removeEvent: {
          _id: '1',
        },
      },
    },
  },
  {
    request: {
      query: UPDATE_EVENT_MUTATION,
      variables: {
        id: '1',
        title: 'Updated title',
        description: 'This is a new update',
        isPublic: false,
        recurring: false,
        isRegisterable: true,
        allDay: true,
        startDate: '2022-03-18',
        endDate: '2022-03-20',
        location: 'New Delhi',
      },
    },
    result: {
      data: {
        updateEvent: {
          _id: '1',
        },
      },
    },
  },
  {
    request: {
      query: UPDATE_EVENT_MUTATION,
      variables: {
        id: '1',
        title: 'Updated title',
        description: 'This is a new update',
        isPublic: false,
        recurring: false,
        isRegisterable: true,
        allDay: false,
        startDate: '2022-03-18',
        endDate: '2022-03-20',
        location: 'New Delhi',
        startTime: '09:00:00Z',
        endTime: '17:00:00Z',
      },
    },
    result: {
      data: {
        updateEvent: {
          _id: '1',
        },
      },
    },
  },
  {
    request: {
      query: REGISTER_EVENT,
      variables: { eventId: '1' },
    },
    result: {
      data: {
        registerForEvent: [
          {
            _id: '123',
          },
        ],
      },
    },
  },
];

export const ERROR_MOCKS = [
  {
    request: {
      query: DELETE_EVENT_MUTATION,
      variables: {
        id: '1',
      },
    },
    error: new Error('Something went wrong'),
  },
];
