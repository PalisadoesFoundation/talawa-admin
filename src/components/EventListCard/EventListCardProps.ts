import { Frequency, WeekDays } from 'utils/recurrenceUtils';
import type { InterfaceEventListCardProps } from './EventListCard';

export const props: InterfaceEventListCardProps[] = [
  {
    key: '',
    id: '',
    eventLocation: '',
    eventName: '',
    eventDescription: '',
    startDate: '',
    endDate: '',
    startTime: '',
    endTime: '',
    allDay: false,
    recurring: false,
    recurrenceRule: null,
    isRecurringEventException: false,
    isPublic: false,
    isRegisterable: false,
    refetchEvents: (): void => {
      /* refetch function */
    },
  },
  {
    key: '123',
    id: '1',
    eventLocation: 'India',
    eventName: 'Shelter for Dogs',
    eventDescription: 'This is shelter for dogs event',
    startDate: '2022-03-19',
    endDate: '2022-03-26',
    startTime: '02:00',
    endTime: '06:00',
    allDay: false,
    recurring: false,
    recurrenceRule: null,
    isRecurringEventException: false,
    isPublic: true,
    isRegisterable: false,
    refetchEvents: (): void => {
      /* refetch function */
    },
  },
  {
    userRole: 'USER',
    key: '123',
    id: '1',
    eventLocation: 'India',
    eventName: 'Shelter for Dogs',
    eventDescription: 'This is shelter for dogs event',
    startDate: '2022-03-19',
    endDate: '2022-03-26',
    startTime: '02:00',
    endTime: '06:00',
    allDay: true,
    recurring: false,
    recurrenceRule: null,
    isRecurringEventException: false,
    isPublic: true,
    isRegisterable: false,
    creator: {
      firstName: 'Joe',
      lastName: 'David',
      _id: '123',
    },
    registrants: [
      {
        _id: '234',
      },
    ],
    refetchEvents: (): void => {
      /* refetch function */
    },
  },
  {
    userRole: 'USER',
    key: '123',
    id: '1',
    eventLocation: 'India',
    eventName: 'Shelter for Dogs',
    eventDescription: 'This is shelter for dogs event',
    startDate: '2022-03-19',
    endDate: '2022-03-26',
    startTime: '02:00',
    endTime: '06:00',
    allDay: true,
    recurring: false,
    recurrenceRule: null,
    isRecurringEventException: false,
    isPublic: true,
    isRegisterable: false,
    creator: {
      firstName: 'Joe',
      lastName: 'David',
      _id: '123',
    },
    registrants: [
      {
        _id: '456',
      },
    ],
    refetchEvents: (): void => {
      /* refetch function */
    },
  },
  {
    userRole: 'ADMIN',
    key: '123',
    id: '1',
    eventLocation: 'India',
    eventName: 'Shelter for Cats',
    eventDescription: 'This is shelter for cat event',
    startDate: '2022-03-19',
    endDate: '2022-03-19',
    startTime: '2:00',
    endTime: '6:00',
    allDay: false,
    recurring: true,
    recurrenceRule: {
      recurrenceStartDate: '2022-03-19',
      recurrenceEndDate: '2022-03-26',
      frequency: Frequency.WEEKLY,
      weekDays: [WeekDays.SATURDAY],
      interval: 1,
      count: null,
      weekDayOccurenceInMonth: null,
    },
    isRecurringEventException: false,
    isPublic: true,
    isRegisterable: false,
    refetchEvents: (): void => {
      /* refetch function */
    },
  },
  {
    userRole: 'ADMIN',
    key: '123',
    id: '1',
    eventLocation: 'India',
    eventName: 'Shelter for Cats',
    eventDescription: 'This is shelter for cat event',
    startDate: '2022-03-17',
    endDate: '2022-03-17',
    startTime: null,
    endTime: null,
    allDay: true,
    recurring: true,
    recurrenceRule: {
      recurrenceStartDate: '2022-03-17',
      recurrenceEndDate: null,
      frequency: Frequency.MONTHLY,
      weekDays: [WeekDays.THURSDAY],
      interval: 1,
      count: null,
      weekDayOccurenceInMonth: 3,
    },
    isRecurringEventException: false,
    isPublic: true,
    isRegisterable: false,
    refetchEvents: (): void => {
      /* refetch function */
    },
  },
  {
    userRole: 'ADMIN',
    key: '123',
    id: '1',
    eventLocation: 'India',
    eventName: 'Shelter for Cats',
    eventDescription: 'This is shelter for cat event',
    startDate: '2022-03-17',
    endDate: '2022-03-17',
    startTime: null,
    endTime: null,
    allDay: true,
    recurring: true,
    recurrenceRule: {
      recurrenceStartDate: '2022-03-17',
      recurrenceEndDate: '2023-03-17',
      frequency: Frequency.MONTHLY,
      weekDays: [WeekDays.THURSDAY],
      interval: 1,
      count: null,
      weekDayOccurenceInMonth: 3,
    },
    isRecurringEventException: false,
    isPublic: true,
    isRegisterable: false,
    refetchEvents: (): void => {
      /* refetch function */
    },
  },
];
