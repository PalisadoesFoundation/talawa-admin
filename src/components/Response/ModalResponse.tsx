import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';

function ModalResponse(props: any) {
  return (
    <Modal show={props.show} onHide={props.handleClose}>
      <Modal.Header closeButton>
        <Modal.Title>Confirmation</Modal.Title>
      </Modal.Header>
      <Modal.Body>{props.message}</Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" onClick={props.handleContinue}>
          Okay
        </Button>
        <Button variant="secondary" onClick={props.handleClose}>
          Cancel
        </Button>
      </Modal.Footer>
    </Modal>
  );
}

export default ModalResponse;
