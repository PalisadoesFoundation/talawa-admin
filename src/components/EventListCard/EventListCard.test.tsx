import React from 'react';
import { render, screen } from '@testing-library/react';
import EventListCard from './EventListCard';
import {
  ApolloClient,
  NormalizedCacheObject,
  ApolloProvider,
  InMemoryCache,
} from '@apollo/client';
import ModalResponse from 'components/Response/ModalResponse';
import userEvent from '@testing-library/user-event';

const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
  cache: new InMemoryCache(),
  uri: 'https://talawa-graphql-api.herokuapp.com/graphql',
});

describe('Testing Event List Card', () => {
  test('should render props and text elements test for the page component', () => {
    render(
      <ApolloProvider client={client}>
        <EventListCard
          key="123"
          id=""
          eventLocation="Gujarat"
          eventName="Shelter for Dogs"
          totalAdmin="5"
          totalMember="107"
          eventImage=""
          regDate="07/04/2020"
          regDays="3"
        />
      </ApolloProvider>
    );
    expect(screen.getByText('Admin:')).toBeInTheDocument();
    expect(screen.getByText('Member:')).toBeInTheDocument();
    expect(screen.getByText('Days:')).toBeInTheDocument();
    expect(screen.getByText('Gujarat')).toBeInTheDocument();
    expect(screen.getByText('Shelter for Dogs')).toBeInTheDocument();
    expect(screen.getByText('5')).toBeInTheDocument();
    expect(screen.getByText('107')).toBeInTheDocument();
    expect(screen.getByText('07/04/2020')).toBeInTheDocument();
    expect(screen.getByText('3')).toBeInTheDocument();
  });

  test('should show modal when the button is clicked', () => {
    render(
      <ApolloProvider client={client}>
        <EventListCard
          key="123"
          id=""
          eventLocation="Gujarat"
          eventName="Shelter for Dogs"
          totalAdmin="5"
          totalMember="107"
          eventImage=""
          regDate="07/04/2020"
          regDays="3"
        />
      </ApolloProvider>
    );
    userEvent.click(screen.getByText('Delete', { selector: 'button' }));
    expect(
      screen.getByText('Are you sure you want to delete this event')
    ).toBeInTheDocument();
  });

  test('should delete event when okay is clicked', async () => {
    render(
      <ApolloProvider client={client}>
        <EventListCard
          key="123"
          id=""
          eventLocation="Gujarat"
          eventName="Shelter for Dogs"
          totalAdmin="5"
          totalMember="107"
          eventImage=""
          regDate="07/04/2020"
          regDays="3"
        />
        <ModalResponse
          show={true}
          message=""
          handleClose={() => { }}
          handleContinue={() => { }}
        />
      </ApolloProvider>
    );
    userEvent.click(screen.getByText('Okay', { selector: 'button' }));
    expect(
      await screen.queryByText('Are you sure you want to delete this event')
    ).toBeNull();
  });
});
