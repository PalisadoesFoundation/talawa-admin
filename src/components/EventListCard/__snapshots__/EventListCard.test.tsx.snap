// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Event List Card should delete event when okay is clicked 1`] = `
<body
  class="modal-open"
  style="padding-right: 0px;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="Toastify"
    />
    <div
      class="orglist row"
    >
      <img
        class="orgimg"
        src="https://via.placeholder.com/200x100"
      />
      <div
        class="singledetails col"
      >
        <div
          class="singledetails_data_left"
        >
          <p
            class="orgname"
          >
            <p>
              Shelter for Dogs
            </p>
          </p>
          <p
            class="orgfont"
          >
            <p>
              Gujarat
            </p>
          </p>
          <p
            class="orgfont"
          >
            Admin: 
            <span>
              5
            </span>
          </p>
          <p
            class="orgfont"
          >
            Member: 
            <span>
              107
            </span>
          </p>
        </div>
        <div
          class="singledetails_data_right"
        >
          <p
            class="orgfont"
          >
            Days: 
            <span>
              3
            </span>
          </p>
          <p
            class="orgfont"
          >
            On: 
            <span>
              07/04/2020
            </span>
          </p>
          <button
            class="orgfontcreatedbtn"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <hr />
  </div>
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal show"
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title h4"
          >
            Confirmation
          </div>
          <button
            class="close"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
            <span
              class="sr-only"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="modal-body"
        />
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            type="button"
          >
            Okay
          </button>
          <button
            class="btn btn-secondary"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
